src/Bicep.Core/LanguageConstants.cs

```diff
@@ -140,6 +145,8 @@ public static class LanguageConstants
         public static readonly TypeSymbol Bool = new PrimitiveType("bool", TypeSymbolValidationFlags.Default);
         public static readonly TypeSymbol Null = new PrimitiveType(NullKeyword, TypeSymbolValidationFlags.Default);
         public static readonly TypeSymbol Array = new ArrayType("array");
+        //Type for available loadTextContent encoding
+        public static readonly TypeSymbol LoadTextContentEncodings = UnionType.Create(Encoding.GetEncodings().OrderBy(e => e.Name).Select(e => new StringLiteralType(e.Name)));
```

Comment:
Can the list of encodings vary by build target? I tested this out on .NET 4.7.2 and on .NET 5, and I get different results:
https://dotnetfiddle.net/GMso4p
https://dotnetfiddle.net/KHUoJx

Personally I feel that we should start by being very controlled and specific about the set of encodings we want to support (rather than just returning the full list of platform supported encodings), and then expand that list if/when we get user feedback about it.

Reply:
Yes, .NET4 has much more available encodings.
I think we should limit them to only: us-ascii, iso-8859-1, utf-8, utf-16BE, utf-16LE, which are available on both platforms (https://docs.microsoft.com/en-us/dotnet/api/system.text.encoding?view=net-5.0#remarks)

Reply:
This is a good point. We definitely should control the list. the `utf-8` expects the BOM, so we may also need to include a UTF-8 version without BOM. We can defer it until we get an ask for it. Adding a new encoding is safe from a back-compat perspective.

Reply:
@majastrz @anthony-c-martin - I just started to wonder - why we want to load utf-16 files, if they will be converted to utf-8 anyway, as we output json files as utf8, so if someone decides to use utf-16 special characters, they will be lost during the deployment anyway...

Reply:
Stream Reader which I use for loading files does not take separate encodings for UTF-8 BOM/no-BOM. It just takes UTF-8 encoding. However if file has a BOM, and you provide a different encoding - you will get a warning and file will be loaded with UTF8. if file does not have BOM and you provide invalid encoding - StreamReader will try to read it as UTF-8 resulting in invalid (from user perspective) result that we cannot detect.
I added negative and positive tests for it.